ruby.js

```diff
@@ -104,6 +107,46 @@ function balanceEvent(event) {
 	if (event && event.document) balancePairs(event.document);
 }
 
+var completeCmd;
+
+function completionProvider(document, position, token) {
+	return new Promise((resolve, reject) => {
+		const line = position.line + 1;
+		const column = position.character;
+
+		var child = cp.spawn(completeCmd, [
+			'--completion-class-info',
+			'--dev',
+			'--fork',
+			'--line='+line,
+			'--column='+column], { detached: false });
+		child.stderr.on('data', (data) => {
+			console.log("stderr:", data.toString());
+			reject(data);
+		});
+		child.stdout.on('data', (data) => {
+			var completionItems = [];
```

Comment:
Collect the completionItems in the scope, but handle the response in `.on('end'` instead.